src/types.md

```diff
@@ -0,0 +1,183 @@
+# Questions about your types
+
+## My 'class' needs references to lots of other things in order to do its job, and that conflicts with other things which also hold similar references. What do I do?
+
+It's common in C++ to have objects which need to mutate other objects to get
```

Comment:
Maybe delete "need to"? It's shorter, and arguably not all instances of this pattern need to do it this way -- many of them could probably be refactored in the way you describe below, it's just that C++ code often chooses the path of "have the object hold onto everything it needs" because it's possible and convenient.